server/interpreter.js

```diff
@@ -2048,6 +2047,19 @@ Interpreter.prototype.executeException = function(error) {
   throw realError;
 };
 
+/**
+ * Create and push a new state onto the statestack.
+ * @param {!Object} node AST node for the state.
+ * @return {!Object} New state.
+ */
+Interpreter.prototype.pushNode_ = function(node) {
```

Comment:
That would mean that `this` in that function will refer to something other than where the function is written.  It will be all alone with no other functions behaving like this.  The function will be well-hidden from IDEs and programmers alike, discoverable only by tracing the interpreter's initialization code.  I think this is the first time I've seen monkey patching suggested for improving clarity.

Also, the signature is just `this.pushNode_(node);`, stateStack is never mentioned in the call.